test_runner/regress/test_physical_replication.py

```diff
@@ -18,12 +18,15 @@ def test_physical_replication(neon_simple_env: NeonEnv):
                 )
         time.sleep(1)
         with env.endpoints.new_replica_start(origin=primary, endpoint_id="secondary") as secondary:
-            with primary.connect() as p_con:
-                with p_con.cursor() as p_cur:
-                    with secondary.connect() as s_con:
-                        with s_con.cursor() as s_cur:
-                            for pk in range(n_records):
-                                p_cur.execute("insert into t (pk) values (%s)", (pk,))
-                                s_cur.execute(
-                                    "select * from t where pk=%s", (random.randrange(1, n_records),)
-                                )
+            p_con = primary.connect()
+            s_con = secondary.connect()
+            count = 0
+            with p_con.cursor() as p_cur:
+                with s_con.cursor() as s_cur:
+                    for pk in range(n_records):
+                        p_cur.execute("insert into t (pk) values (%s)", (pk,))
+                        s_cur.execute(
+                            "select count(*) from t where pk=%s", (random.randrange(1, n_records),)
+                        )
+                        count += s_cur.fetchall()[0][0]
+            assert count > 0
```

Comment:
Even sync replication can not guarantee that client will see previous insert on replica...
But in case of this test select uses random key not correlated somehow with insert key.
So actually this test is flaky (this is non-zero probability that we never lockup key which was already inserted).